### PR TITLE
chore(ui): migrate PortalProvider to Typescript

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

migrate PortalProvider to Typescript

# Changes Made

- migrate PortalProvider to Typescript
- temporarily use mock Message in tests

# Related Issues

- Issue 1: [link to issue](https://github.com/cloudoperators/juno/issues/245)

# Testing Instructions

1. `npm i`
2. `turbo test`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/components/PortalProvider/PortalProvider.component.tsx
Comment (andypf) [inline packages/ui-components/src/components/PortalProvider/PortalProvider.component.tsx:141]: missing new line

**Review by andypf — COMMENTED** [review]
Looks good 👍 , but please fix the formatting and ensure all tests pass.

**Review by andypf — DISMISSED** [review]

**Review by andypf — APPROVED** [review]

File: packages/ui-components/src/components/PortalProvider/PortalProvider.stories.tsx
Comment (barsukov) [inline packages/ui-components/src/components/PortalProvider/PortalProvider.stories.tsx:10]: Lets wait until message component merge then we actualize it

**Review by barsukov — CHANGES_REQUESTED** [review]
This PR is blocked by message PR.

File: packages/ui-components/src/components/PortalProvider/PortalProvider.stories.tsx
Comment (gjaskiewicz) [reply packages/ui-components/src/components/PortalProvider/PortalProvider.stories.tsx]: I changed to Button as discussed during the meeting.

**Review by andypf — APPROVED** [review]

**Pull request merged**
